File: src/main/resources/output/IFC4X3_js/Entities/IfcStructuralLoad/IfcStructuralLoadOrResult/IfcStructuralLoadStatic/IfcStructuralLoadSingleDisplacement/IfcStructuralLoadSingleDisplacement.js
class IfcStructuralLoadSingleDisplacement extends IfcStructuralLoadStatic {
    constructor() {
        /** @type {IFCLENGTHMEASURE} */
        this.DisplacementX = null;
        /** @type {IFCLENGTHMEASURE} */
        this.DisplacementY = null;
        /** @type {IFCLENGTHMEASURE} */
        this.DisplacementZ = null;
        /** @type {IFCPLANEANGLEMEASURE} */
        this.RotationalDisplacementRX = null;
        /** @type {IFCPLANEANGLEMEASURE} */
        this.RotationalDisplacementRY = null;
        /** @type {IFCPLANEANGLEMEASURE} */
        this.RotationalDisplacementRZ = null;
    }

    // === EXTENDED BY ===
    // IfcStructuralLoadSingleDisplacementDistortion
}

File: src/main/resources/output/IFC4X3_csharp/Entities/IfcStructuralLoad/IfcStructuralLoadOrResult/IfcStructuralLoadStatic/IfcStructuralLoadSingleDisplacement/IfcStructuralLoadSingleDisplacement.cs
public class IfcStructuralLoadSingleDisplacement : IfcStructuralLoadStatic
{
    public IfcLengthMeasure DisplacementX { get; set; }
    public IfcLengthMeasure DisplacementY { get; set; }
    public IfcLengthMeasure DisplacementZ { get; set; }
    public IfcPlaneAngleMeasure RotationalDisplacementRX { get; set; }
    public IfcPlaneAngleMeasure RotationalDisplacementRY { get; set; }
    public IfcPlaneAngleMeasure RotationalDisplacementRZ { get; set; }

    // === EXTENDED BY ===
    // IfcStructuralLoadSingleDisplacementDistortion
}
